mesonbuild/mconf.py

```diff
@@ -77,6 +79,33 @@ def __init__(self, build_dir: str):
             self.source_dir = self.build.environment.get_source_dir()
             self.coredata = self.build.environment.coredata
             self.default_values_only = False
+
+            # if the option file has been updated, reload it
+            # This cannot handle options for a new subproject that has not yet
+            # been configured.
+            for sub, options in self.coredata.options_files.items():
+                if options is not None and os.path.exists(options[0]):
+                    opfile = options[0]
+                    with open(opfile, 'rb') as f:
+                        ophash = hashlib.sha1(f.read()).hexdigest()
```

Comment:
why do you use sha1 here, and md5 in interpreter.py?

Reply:
Oops, originally I had used md5 everywhere, but then read the docs that you can compile md5 out of Python, but not sha1. I'll switch to sha1 everywhere.